neurodamus/core/stimuli.py

```diff
@@ -465,37 +487,32 @@ def __init__(
             cell_section,
             position=0.5,
             clamp_container=None,
-            stim_vec_mode=True,
             time_vec=None,
             stim_vec=None,
             reversal=0.0,
-            **clamp_params,
+            represents_physical_electrode=False,
         ):
-            represents_physical_electrode = clamp_params.get("represents_physical_electrode", False)
-            # Checks if new conductanceSource mechanism is available and if source does not
-            # represent physical electrode, otherwise fall back to SEClamp.
-            if not represents_physical_electrode and hasattr(Neuron.h, "ConductanceSource"):
-                self.clamp = Neuron.h.ConductanceSource(position, sec=cell_section)
-            else:
-                self.clamp = Neuron.h.SEClamp(position, sec=cell_section)
+            # source does not represent physical electrode,
+            # otherwise fall back to SEClamp.
+            self.clamp = (
+                Nd.h.SEClamp(position, sec=cell_section)
+                if represents_physical_electrode
+                else Nd.h.ConductanceSource(position, sec=cell_section)
+            )
 
-            if stim_vec_mode:
-                assert time_vec is not None
-                assert stim_vec is not None
-                self.clamp.dur1 = time_vec[-1]
-                self.clamp.amp1 = reversal
-                # support delay with initial zero
-                self.time_vec = Neuron.h.Vector(1, 0).append(time_vec)
-                self.stim_vec = Neuron.h.Vector(1, 0).append(stim_vec)
-                # replace self.stim_vec with inverted and clamped signal
-                # rs is in MOhm, so conductance is in uS (micro Siemens)
-                self.stim_vec = Neuron.h.Vector(
-                    [1 / x if x > 1e-9 and x < 1e9 else 1e9 for x in self.stim_vec]
-                )
-                self.stim_vec.play(self.clamp._ref_rs, self.time_vec, 1)
-            else:
-                for param, val in clamp_params.items():
-                    setattr(self.clamp, param, val)
+            assert time_vec is not None
```

Comment:
Is it better to keep the `if-else` condition for the future implementation ?

Reply:
atm there is no way to reach that point (except ad hoc examples where we use _* methods) as the only function that instantiate these objects is `attach_to`. I would things as lean as possible. We can easily re-add if the feature is needed